openshift-kube-apiserver/admission/customresourcevalidation/route/validate_route.go

```diff
@@ -45,18 +41,21 @@ func toRoute(uncastObj runtime.Object) (*routev1.Route, field.ErrorList) {
 }
 
 type routeV1 struct {
+	sarc           authorizationv1client.SubjectAccessReviewInterface
```

Comment:
BTW: Perhaps this question should be asked elsewhere. But could you explain why we need the `SAR` client? It seems to me that we want to make sure if the user can change `Route.Spec.Host`, right? Do you happen to know why this was implemented in this particular way?

Reply:
/cc @benluddy

Reply:
The SAR client is specifically needed by the host update validation (https://github.com/openshift/library-go/blob/c91dd9756953b9b7ff3139565886954ed0cc33d3/pkg/route/hostassignment/assignment.go#L120). It is used to verify rbac on the `custom-host` sub resource. I am re-using the SAR client to also validate permissions needed for reading secrets (https://github.com/openshift/enhancements/blob/master/enhancements/ingress/route-secret-injection-for-external-certificate-management.md#implementation-detailsnotesconstraints-optional).